File: src/main/java/com/graqr/threshr/model/queryparam/TargetStore.java
package com.graqr.threshr.model.queryparam;

import io.micronaut.core.annotation.Introspected;
import io.micronaut.http.annotation.QueryValue;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
@Introspected
public class TargetStore {
    @QueryValue("store_id")
    String storeId;

    @QueryValue("scheduled_delivery_store_id")
    String scheduledDeliveryStoreId;

    @QueryValue("required_store_id")
    String requiredStoreId;
    String zip, state;
    double latitude, longitude;

    public TargetStore(String locationId, String region, String postalCode, double latitude, double longitude) {
        storeId = locationId;
        scheduledDeliveryStoreId = locationId;
        requiredStoreId = locationId;
        zip = postalCode;
        this.latitude = latitude;
        this.longitude = longitude;
        state = region;
    }
}

File: src/main/kotlin/com/graqr/threshr/circleStore/model/queryparam/TargetStore.kt
package com.graqr.threshr.circleStore.model.queryparam

import io.micronaut.core.annotation.Introspected
import io.micronaut.http.annotation.QueryValue

/**
 * TargetStore object used with micronaut's deserialization for querying the RedSky API
 */
@Introspected
data class TargetStore(
    @param:QueryValue("store_id") val storeId: String,
    @param:QueryValue("scheduled_delivery_store_id") val scheduledDeliveryStoreId: String,
    @param:QueryValue("required_store_id") val requiredStoreId: String,
    val zip: String, val state: String, val latitude: Double, val longitude: Double) {

    /**
     * instantiates a TargetStore where the locationID string is used for store_id, scheduled_delivery_store_id,
     * and required_store_id
     */
    constructor(locationId: String, region: String, postalCode: String, latitude: Double, longitude: Double) :
            this(locationId, locationId, locationId, postalCode, region, latitude, longitude) {}
}
